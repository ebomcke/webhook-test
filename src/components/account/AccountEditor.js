import React, { Component } from 'react';
import { compose } from 'recompose';
import { withFirestore } from 'react-firestore';
import { withFirebase } from '../../contexts/Firebase';
import { withAccount } from '../../contexts/Account';

class AccountEditor extends Component {
  constructor(props) {
    super(props);
    let { account } = this.props;
    this.state = account;
  }

  handleInputChange = ({ event, name, data }) => {
    const target = event.target;
    let value;
    if (data) {
      value = data.value;
    } else {
      value = target.type === 'checkbox' ? target.checked : target.value;
    }
    if (!name) {
      name = target.name;
    }

    this.setState({
      [name]: value,
    });
  }

  save = async () => {
    const { firebase, firestore } = this.props;
    if (!this.state.confirmed) {
      this.state.confirmed = firebase.firestore.Timestamp.now();
    }
    await firestore
      .collection('accounts')
      .doc(this.state.id)
      .set(this.state);
    return this.state.id;
  }

  render() {
    const { children } = this.props;
    return children({
      account: this.state,
      handleInputChange: this.handleInputChange,
      save: this.save,
    });
  }
}

export default compose(
  withFirebase,
  withFirestore,
  withAccount
)(AccountEditor);
