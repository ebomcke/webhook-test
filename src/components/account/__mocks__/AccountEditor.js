const mockAccountEditor = {
    account: {
        name: 'Unit Test',
        organisationName: 'unit-test-org-name'
    },
    onOrganisationNameChange: jest.fn(),
    save: jest.fn(() => Promise.resolve())
}

export { mockAccountEditor };