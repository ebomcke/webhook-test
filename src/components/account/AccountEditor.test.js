import React from 'react';
import AccountEditor from './AccountEditor';
import renderer from 'react-test-renderer';

describe('Rendering tests', () => {
  test('Child inherits relevant props', () => {
    const accountEditor = renderer.create(
      <AccountEditor>
        {accountEditor => <div accountEditor={accountEditor} />}
      </AccountEditor>,
    );
    let tree = accountEditor.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Interactions tests', () => {
  test('onOrganisationNameChange updates state value', () => {
    const accountEditor = renderer.create(
      <AccountEditor>
        {accountEditor => <div hasEditor={true} accountEditor={accountEditor} />}
      </AccountEditor>,
    );

    accountEditor.root.findByProps({hasEditor:true}).props.accountEditor.onOrganisationNameChange('test-value');
    let tree = accountEditor.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
