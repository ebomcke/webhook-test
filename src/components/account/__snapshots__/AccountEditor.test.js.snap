// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interactions tests onOrganisationNameChange updates state value 1`] = `
<div
  accountEditor={
    Object {
      "account": Object {
        "organisationName": "test-value",
      },
      "onOrganisationNameChange": [Function],
      "save": [Function],
    }
  }
  hasEditor={true}
/>
`;

exports[`Rendering tests Child inherits relevant props 1`] = `
<div
  accountEditor={
    Object {
      "account": Object {
        "organisationName": "unit-test-org",
      },
      "onOrganisationNameChange": [Function],
      "save": [Function],
    }
  }
/>
`;
